syft/frameworks/torch/nn/rnn.py

```diff
@@ -309,34 +280,24 @@ def _init_hidden(self, input):
                 h = h.share(*owners, crypto_provider=crypto_provider)
         return h
 
-    def _apply_time_step(self, x, h, c, t, reverse_direction=False):
+    def _apply_time_step(self, x, h, t, reverse_direction=False):
         """
         Apply RNN layers at time t, given input and previous hidden states
         """
         rnn_layers = self.rnn_backward if reverse_direction else self.rnn_forward
 
-        h_next = h.new(h.shape).zero_()
-        c_next = c.new(c.shape).zero_() if self.is_lstm else None
+        h = torch.stack([*h])
+        h_next = torch.stack([item.new(item.shape).zero_() for item in h])
```

Comment:
I'm not sure what happens here but I don't think it's the easiest way to do it... Is it?

Reply:
(recall `h` is now `hc`, per previous comment)
`torch.stack([*hc])` just ensures that `hc` is a tensor, not a list.
And no, the second line is convoluted per my blindly direct translation of the previous code. I've now replaced it with: `hc_next = torch.zeros_like(hc)`